src/common/watchdog.cpp

```diff
@@ -34,11 +34,12 @@ Watchdog::~Watchdog()
 {
     if (m_running)
     {
-        std::unique_lock<std::mutex> lock(m_bottlneck);
+        m_bottlneck.lock();
```

Comment:
Can we achieve this same thing with some careful scoping? Manual lock/unlock is bad juju in my experience